### PR TITLE
[NUI] Layouting made public

### PR DESCRIPTION
**Pull request by dongsug-song**
### Description of Change ###
Description of Change
NUI Layouting API made public
LayoutGroup _children property renamed to LayoutChildren
Classes that were previously marked internal now public.
Change-Id: I78b0a54c07d69496a666740fd6e09ade1dd6bc9b

### API Changes ###
http://suprem.sec.samsung.net/jira/browse/TCSACR-278

### CODE REVIEW

File: src/Tizen.NUI/src/public/Layouting/MeasureSpecification.cs [hunk 1]
```diff
@@ -24,29 +24,36 @@ namespace Tizen.NUI
...
         /// MeasureSpecification Mode.
         /// </summary>
+        /// <since_tizen> 6 </since_tizen>
         public MeasureSpecification.ModeType Mode;
 
         /// <summary>
         /// Constructor taking size and mode type.
         /// </summary>
         /// <param name="size">size value.</param>
         /// <param name="mode">mode vaue.</param>
```
Comment (m-macias) [inline src/Tizen.NUI/src/public/Layouting/MeasureSpecification.cs]: Spelling.
```suggestion
        /// <param name="mode">mode value.</param>
```

File: src/Tizen.NUI/src/public/Layouting/MeasuredSize.cs [hunk 2]
```diff
@@ -40,6 +41,7 @@ public MeasuredSize(LayoutLength measuredSize, MeasuredSize.StateType state)
         /// Creates a MeasuredSize from a LayoutLength
         /// </summary>
         /// <param name="measuredSize">LayoutLength to create </param>
+        /// <since_tizen> 6 </since_tizen>
         public static implicit operator MeasuredSize(LayoutLength measuredSize)
```
Comment (m-macias) [inline src/Tizen.NUI/src/public/Layouting/MeasuredSize.cs]: Methods which return non-void value must have exactly one 'returns' section.

File: src/Tizen.NUI/src/public/Layouting/LayoutTransition.cs [hunk 3]
```diff
@@ -17,56 +17,94 @@
...
+
+        Remove = 4,
+        /// <summary>
+        /// Animation when an item changes due to a sibbling being added.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+
+        ChangeOnAdd = 8,
+        /// <summary>
+        /// Animation when an item changes due to a sibbling being removed.
```
Comment (m-macias) [inline src/Tizen.NUI/src/public/Layouting/LayoutTransition.cs]: Spelling.
```suggestion
        /// Animation when an item changes due to a sibling being removed.
```

File: src/Tizen.NUI/src/public/Layouting/LayoutTransition.cs [hunk 4]
```diff
@@ -17,56 +17,94 @@
...
+
+        Add = 2,
+        /// <summary>
+        /// Animate removing item.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+
+        Remove = 4,
+        /// <summary>
+        /// Animation when an item changes due to a sibbling being added.
```
Comment (m-macias) [inline src/Tizen.NUI/src/public/Layouting/LayoutTransition.cs]: Spelling.
```suggestion
        /// Animation when an item changes due to a sibling being added.
```

File: src/Tizen.NUI/src/public/Layouting/LayoutTransition.cs [hunk 5]
```diff
@@ -92,35 +130,30 @@ public TransitionComponents(int delay, int duration, AlphaFunction alphaFunction
...
         /// </summary>
-        /// Hidden-API which is usually used as Inhouse-API. If required to be opened as Public-API, ACR process is needed.
-        [EditorBrowsable(EditorBrowsableState.Never)]
+        /// <since_tizen> 6 </since_tizen>
+
         public AlphaFunction AlphaFunction;
     }
 
     /// <summary>
     /// LayoutTransition stores the aninmation setting for a transition conidition.
```
Comment (m-macias) [inline src/Tizen.NUI/src/public/Layouting/LayoutTransition.cs]: Spelling.
```suggestion
    /// LayoutTransition stores the animation setting for a transition condition.
```

File: src/Tizen.NUI/src/public/Layouting/LayoutTransition.cs [hunk 6]
```diff
@@ -80,8 +118,8 @@ public TransitionComponents()
         /// <param name="delay">The delay before the animator starts.</param>
         /// <param name="duration">the duration fo the animator.</param>
```
Comment (m-macias) [inline src/Tizen.NUI/src/public/Layouting/LayoutTransition.cs]: Spelling.
```suggestion
        /// <param name="duration">the duration of the animator.</param>
```

File: src/Tizen.NUI/src/public/Layouting/LayoutGroup.cs [hunk 7]
```diff
@@ -65,30 +72,32 @@ public virtual void Add(LayoutItem childLayout)
...
+            LayoutChildren.Clear();
             // todo ensure child LayoutItems are still not parented to this group.
             RequestLayout();
         }
 
         /// <summary>
         /// From ILayoutParent
         /// </summary>
+        /// <since_tizen> 6 </since_tizen>
         public virtual void Remove(LayoutItem layoutItem)
```
Comment (m-macias) [inline src/Tizen.NUI/src/public/Layouting/LayoutGroup.cs]: Parameters of method must have valid documentation. Method has 1 parameter, 0 are documented.

File: src/Tizen.NUI/src/public/Layouting/LayoutGroup.cs [hunk 8]
```diff
@@ -26,33 +26,40 @@ namespace Tizen.NUI
...
-            _children = new List<LayoutItem>();
+            LayoutChildren = new List<LayoutItem>();
         }
 
-          /// <summary>
+        /// <summary>
         /// From ILayoutParent.<br />
         /// </summary>
+        /// <since_tizen> 6 </since_tizen>
         public virtual void Add(LayoutItem childLayout)
```
Comment (m-macias) [inline src/Tizen.NUI/src/public/Layouting/LayoutGroup.cs]: Parameters of method must have valid documentation. Method has 1 parameter, 0 are documented.

**Comment by agnelovaz** [comment]
Created new pull request to update with latest changes - https://github.com/Samsung/TizenFX/pull/1026

**Pull request closed without merging**